openmmtools/cache.py

```diff
@@ -459,7 +481,13 @@ def _generate_state_id(thermodynamic_state):
         """Return a unique key for the ThermodynamicState."""
         # We take advantage of the cached _standard_system_hash property
         # to generate a compatible hash for the thermodynamic state.
-        return str(thermodynamic_state._standard_system_hash)
+        return thermodynamic_state._standard_system_hash
+
+    @classmethod
+    def _generate_integrator_id(cls, integrator):
+        """Return a unique key for the given Integrator."""
+        standard_integrator = cls._standardize_integrator(integrator)
+        return openmm.XmlSerializer.serialize(standard_integrator).__hash__()
```

Comment:
So I think the situation here is the same as for the states classes. This hash is only kept by the `ContextCache` object, which is unique for each process, and it's not related to the hash computed by `RestorableIntegrator`. So the situation is a little different from `RestorableIntegrator` where the hash becomes part of the state of the integrator (i.e. it's part of its serialization) and can be send around to multiple processes.

If we do end up converting everything to deterministic algorithm, we should use a 64-bit hash. I've used `adler32` in `RestorableIntegrator` only because we need to hash strings of few tens of characters (the names of the integrators).